### ISSUE TITLE
`flash-msg.js` may not work correctly when there is no flash message HTML

### ISSUE THREAD

**Issue by delphidabbler** [delphidabbler/delphidabbler.github.io#130]
Experimentation with cookies showed a cookie name of `dd_flash_undefined` was being tested for.

This implies that perhaps the code that detects for the absence of the `flash-popup` id is not working. Possibly the following test never evaluates true:

https://github.com/delphidabbler/delphidabbler.github.io/blob/881bc606a4687295126f650d67a77a4aeb36cbb2/assets/scripts/flash-msg.js#L9

`flash` is set in the previous line:

https://github.com/delphidabbler/delphidabbler.github.io/blob/881bc606a4687295126f650d67a77a4aeb36cbb2/assets/scripts/flash-msg.js#L8

If certain Jekyll variables are set then the flash HTML should not be present and so the `flash-popup` id should not be present.

The only way I can see that a `dd_flash_undefined` cookie name can be built is if `flash.data('flash-id')` returns undefined in the following code:

```javascript
var id = flash.data('flash-id');
var cookieName = 'dd_flash_' + id;
```

Getting to this code should be impossible when there is no HTML.

So, this code needs to be checked.

**Comment by delphidabbler** [comment]
I misunderstood how jQuery notifies a selector it doesn't find. I assumes `$("#flash-popup")` would return falsy if `#flash-popup` was not found.

What actually happens is that a valid object is returned with zero length.

So need to change the test:

```javascript
if ( ! flash )
```

to

```javascript
if ( flash.length === 0 )
```

**Comment by delphidabbler** [comment]
Will actually fix this bug while implementing fix required by issue #126.
